src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -656,17 +656,13 @@ def _save_checkpoint(
             self.sg_logger.add_checkpoint(tag="ckpt_latest_weights_only.pth", state_dict={"net": self.net.state_dict()}, global_step=epoch)
             return
 
-        # COMPUTE THE CURRENT metric
-        # IF idx IS A LIST - SUM ALL THE VALUES STORED IN THE LIST'S INDICES
-        curr_tracked_metric = float(validation_results_dict[self.metric_to_watch])
+        # Check whether we have to attempt the validation results
+        # and subtract 1 from epoch because epoch+1 is passed to this function
+        is_validation_calculated = self._if_need_to_calc_validation(epoch - 1)
+        curr_tracked_metric = None
 
-        # create metrics dict to save
-        valid_metrics_titles = get_metrics_titles(self.valid_metrics)
```

Comment:
I think this is definitely not the right place where to put this logic TBH.
I think add_checkpoint must be always called when metrics are present. IMO it violates SRP that `add_checkpoint ` must know of such think as validation frequency. 

Instead, the condition we use to trigger `_test_epoch` should be changed instead:

`if (epoch + 1) % self.run_validation_freq== 0` -> `if self._should_run_validation_for_epoch(epoch)` should be used.